operations/lamport/clock.go

```diff
@@ -14,29 +14,33 @@
 
 package lamport
 
-import (
-	"sync/atomic"
-)
+import "sync/atomic"
 
 type Clock struct {
+	// counter represents current lamport counter value.
 	counter uint64
+	// hostId identifies the host where the operation was generated at first place.
 	hostId  string
 }
 
-func (clock *Clock) IsGreaterThan(obj *Clock) bool {
-	if atomic.LoadUint64(&obj.counter) == atomic.LoadUint64(&clock.counter) {
-		return clock.hostId > obj.hostId
-	} else {
-		return atomic.LoadUint64(&clock.counter) > atomic.LoadUint64(&obj.counter)
+// NewClock returns a pointer to a newly created clock.
+func NewClock(counter uint64, hostId string) *Clock {
+	return &Clock{
+		hostId: hostId,
+		counter: counter,
 	}
 }
 
+func (clock *Clock) IsGreaterThan(obj *Clock) bool {
+	return atomic.LoadUint64(&clock.counter) > atomic.LoadUint64(&obj.counter)
+}
+
 func (clock *Clock) IsLessThan(obj *Clock) bool {
-	if atomic.LoadUint64(&obj.counter) == atomic.LoadUint64(&clock.counter) {
-		return clock.hostId < obj.hostId
-	} else {
-		return atomic.LoadUint64(&clock.counter) < atomic.LoadUint64(&obj.counter)
```

Comment:
revert

Reply:
(counter, hostId) defines total ordering of operations across all replicas

Reply:
Just think logically and tell me, how does it differ when the two operations happing at two different hosts have the same Lamport counter. The ordering of the host id should matter? No. Either the host id is auto-generated at init or randomly given during bootstrapping. So why should we take into consideration programming logic?

Reply:
To resolve concurrent operations, For LWW to work we will need someone to win.

Reply:
Yea, I understand. We need to think about this constructively...instead of in such a destructive way like the last writer wins. As CRDT is all about resolving conflict optimistically : )
The topic for discussion on next meet.

Reply:
what if both have random ids which are coincidentally equal. We'll end up introducing randomness into the system

Reply:
each replica **must** have a unique id

Reply:
you can't ensure must. Instead we could focus here
```
Another frequently-used approach to conflict resolution
is last writer wins (LWW), which arbitrarily chooses one
among several concurrent writes as “winner” and discards
the others. LWW is used in Apache Cassandra, for example.
It does not meet our requirements, since we want no user
input to be lost due to concurrent modifications.
```